### PR TITLE
#456 BkReuse removed. Tests from BkReuseTest moved to BkBasicTest

### PR DESCRIPTION
**Pull request by xupyprmv**
#456. I removed `BkReuse` and `BkReuseTest`. All tests from `BkReuseTest` moved to `BkBasicTest`. Tests are still marked as ignored and must be refactored(#454).

### CODE REVIEW

**Comment by davvd** [comment]
@xupyprmv Many thanks for the PR, let me find a reviewer for it

**Comment by davvd** [comment]
@dmzaytsev please review,thanks

**Comment by dmzaytsev** [comment]
@xupyprmv according the original ticket 
>Let's get rid of BkReuse, move all tests from BkReuseTest to BkBasicTest and implement persistent connections inside BkBasic.

Let's add a puzzle to `and implement persistent connections inside BkBasic.`

**Comment by xupyprmv** [comment]
@dmzaytsev 

> Let's add a puzzle to and implement persistent connections inside BkBasic.

according the original ticket (bottom of discussion)

>@HDouss I see next line inside issue description:
implement persistent connections inside BkBasic
Is that comment related to current issue?

>@xupyprmv No. It is a suggestion to let #418 as it is and not to invalidate it (but may be update it).

So, I'm not sure that we create something new. We need just update #418. Add something like:

> implement persistent connections inside `BkBasic` instead of `BkReuse`. (#456)

Do you still think that we need to add new puzzle?

File: src/test/java/org/takes/http/BkReuseTest.java [hunk 1]
```diff
@@ -1,143 +0,0 @@
...
-import org.takes.tk.TkText;
-
-/**
- * Test case for {@link BkReuse}.
- *
- * @author Piotr Pradzynski (prondzyn@gmail.com)
- * @version $Id$
- * @checkstyle MultipleStringLiteralsCheck (500 lines)
- * @todo #306:30min Implement missing BkReuse.accept method to support
- *  HTTP persistent connections and to pass below three tests. BkReuse.accept
```
Comment (dmzaytsev) [inline src/test/java/org/takes/http/BkReuseTest.java]: @xupyprmv you removed the puzzle here that's why I propose to create a new one

**Comment by dmzaytsev** [comment]
@xupyprmv anyway let's ask the  author
@HDouss what do you think about the puzzle?

**Comment by HDouss** [comment]
@dmzaytsev On one hand, the puzzle have to be updated (or a new one created). On the other hand, reference to initial issue #306 is important, and the puzzle and associated issue (#418) are still relevant. I suggest to let the initial puzzle and only remove the `BkReuse.accept` reference.

**Comment by xupyprmv** [comment]
@dmzaytsev I added puzzle with updated description (to avoid confusion)

**Comment by dmzaytsev** [comment]
@xupyprmv thank you!

**Comment by dmzaytsev** [comment]
@rultor merge

**Comment by yegor256** [comment]
@rultor merge

**Comment by xupyprmv** [comment]
@yegor256 I have to update puzzle description. Am I right?

**Comment by yegor256** [comment]
@rultor try to merge

**Comment by HDouss** [comment]
@xupyprmv You have to make a longer description for the puzzle.

**Comment by xupyprmv** [comment]
@yegor256 I updated puzzle description. Please continue with merge.

**Comment by yegor256** [comment]
@rultor try to merge

**Comment by davvd** [comment]
@ypshenychka please, review this issue for QA compliance, as per [par.24](http://at.teamed.io/policy.html#24)

**Comment by davvd** [comment]
@rultor deploy pls

**Comment by ypshenychka** [comment]
@dmzaytsev 
According to our [QA Rules](http://www.teamed.io/qa.html):
>The code reviewer found at least three problems in the code.

Only one issue was found during code review. 
Please confirm that you'll try to find at least three problems while future reviews.

**Comment by dmzaytsev** [comment]
@ypshenychka I confirm

**Comment by ypshenychka** [comment]
@dmzaytsev Thank you

**Comment by ypshenychka** [comment]
@davvd Quality is acceptable.

**Comment by davvd** [comment]
> @davvd Quality is acceptable.

@ypshenychka thanks, got it. everybody, please try to make it good next time

**Comment by davvd** [comment]
@dmzaytsev I added 10 mins to @ypshenychka (for QA review) in transaction 74692502... I just added **21 mins** to your account, many thanks for your contribution (74692519).. 192 hours and 14 mins spent here... review comments (c=6) added as a bonus... +21 added to your rating, at the moment it is: [+1913](http://www.netbout.com/b/35166?open=rating)

**Pull request merged**